Interpolation in makers map

LauraOlivera opened the pull request:
<!-- These are hidden commments. You can leave them or delete them. -->

<!-- Don't be shy! Github issues or pull requests are welcome any time. -->
<!-- But if you first want to chat, use the Gammapy Slack. -->
<!-- See https://gammapy.org/contact.html -->

**Description**
<!-- What is the motivation for this pull request? -->
<!-- Briefly: what changes are done here? -->
<!-- If this is to address a Github issue, metion it's number to create a link -->
<!-- Example: This PR is to fix issue #42 -->

This pull request makes it possible to have a `Map` object as the `aeff` and `bkg` inputs to the data reduction performed by the `MapDatasetMaker`, as well as an `EdispKernelMap` and `PSFMap` for the `edisp` and `psf`. The the input maps are interpolated to the reference geometry before being bundled into the resulting `MapDataset`.

**Dear reviewer**
I included a test that creates a dummy `Map` effective area and background, a dummy `PSFMap`, and `EdispKernelMap` and dummy `GTI` and `Events`. They are bundled into an `Observation` and run through the `MapDatasetMaker` with a given target `geom`. 

<!-- Let the reviewer and Gammapy team know what you want: -->
<!-- * Is this ready for review? Or is it work in progress and you want some feedback? -->
<!-- * Do you want ot go through review here? Or if someone just finish this up and merge it in? -->
<!-- Do you have any specific questions, e.g. about API or implementation? -->
<!-- Do you include a test executing new code you're adding (to make sure it runs)? -->
<!-- Do you include some documentation? Is it needed? -->

adonath commented:
Thanks @LauraOlivera, the test fails on Azure seemed unrelated. I restarted the builds and will do a review later today...

LauraOlivera commented:
@adonath I moved the map interpolation to the `Map` class and replaced it for each IRFs. This way no extra methods are necessary in `makers/utils.py` . I also added a test for the `Map.interp_to_geom()` method.

adonath reviewed (approved):
Thanks a lot @LauraOlivera, I have no further comments..
